openedx/core/djangoapps/theming/management/commands/create_sites_and_configurations.py

```diff
@@ -0,0 +1,180 @@
+                ),
+                "url": url,
+                "client_id": "{client_type}-key-{site_name}".format(
+                    client_type="discovery" if is_discovery else "ecommerce",
+                    site_name=site_name
+                ),
+                "client_secret": "{client_type}-secret-{dns_name}".format(
+                    client_type="discovery" if is_discovery else "ecommerce",
+                    dns_name=self.dns_name
+                ),
+                "client_type": CONFIDENTIAL,
+                "logout_uri": "{url}logout/".format(url=url)
+            }
+        )
+        LOG.info("Adding {client} oauth2 client as trusted client".format(client=client.name))
+        TrustedClient.objects.get_or_create(client=client)
+
+    def _create_sites(self, site_domain, theme_dir_name, site_configuration):
+        """
+        Create Sites, SiteThemes and SiteConfigurations
+        """
+        site, created = Site.objects.get_or_create(
+            domain=site_domain,
+            defaults={"name": theme_dir_name}
+        )
+        if created:
+            LOG.info("Creating '{site_name}' SiteTheme".format(site_name=site_domain))
+            SiteTheme.objects.create(site=site, theme_dir_name=theme_dir_name)
+
+            LOG.info("Creating '{site_name}' SiteConfiguration".format(site_name=site_domain))
+            SiteConfiguration.objects.create(site=site, values=site_configuration, enabled=True)
+        else:
+            LOG.info(" '{site_domain}' site already exists".format(site_domain=site_domain))
+
+    def _get_sites_data(self):
+        """
+        Reads the json file from theme directory and returns the site data in JSON format.
+        """
+        site_data = {}
+        for site_name, site_dir in SITES.items():
+            configuration = json.dumps(
+                json.load(
+                    open("{theme_path}/{site_dir}/sandbox_configuration.json".format(
+                        theme_path=self.theme_path,
+                        site_dir=site_dir
+                    ))
+                )
+            ).replace("{dns_name}", self.dns_name)
+            site_data[site_name] = {
+                "theme_dir_name": site_dir,
+                "configuration": configuration
+            }
+        return site_data
+
+    def _generate_urls(self, site_name):
+        """
+        Generate the lms, discovery and ecommerce URLs for a site.
+        """
+        lms_url = "{site_name}-{dns_name}.sandbox.edx.org".format(site_name=site_name, dns_name=self.dns_name)
+        discovery_url = "https://discovery-{site_name}-{dns_name}.edx.org/".format(
+            site_name=site_name,
+            dns_name=self.dns_name
+        )
+        ecommerce_url = "https://ecommerce-{site_name}-{dns_name}.edx.org/".format(
+            site_name=site_name,
+            dns_name=self.dns_name
+        )
+        if site_name == "edx":
```

Comment:
Why do we have special case if site_name is "edx"? We should use the sandbox_configuration.json to store any variables we need to override so we don't have special case code in here

Reply:
@bill-filler, this is special case. For example for WL-site mitxpro: sandbox url will be 
```mitxpro-{dns_name}.sandbox.org``` but for default edx site: sandbox url can not be ```edx-{dns_name}.sandbox.org```, it should be ```{dns_name}.sandbox.edx.org```